MaeBlock.hpp

```diff
@@ -363,11 +363,7 @@ inline void set_indexed_property(std::map<std::string, std::shared_ptr<T>>& map,
                                  std::shared_ptr<T> value)
```

Comment:
still an extra copy of the shared_ptr created at the call point to `set_indexed_property()`. But I'd assume that any optimizer would eliminate this call.